src/Commands/Command.php

```diff
@@ -44,91 +53,111 @@
- * [[sql]].
+ * {@see sql}.
  * @property string $sql The SQL statement to be executed.
- *
- * @author Qiang Xue <qiang.xue@gmail.com>
- *
- * @since 2.0
  */
-class Command extends Component
+class Command
 {
     /**
-     * @var Connection the DB connection that this command is associated with
+     * @var Connection|null the DB connection that this command is associated with
      */
-    public $db;
+    public ?Connection $db;
+
     /**
-     * @var \PDOStatement the PDOStatement object that this command is associated with
+     * @var PDOStatement|null the PDOStatement object that this command is associated with
      */
-    public $pdoStatement;
+    public ?PDOStatement $pdoStatement = null;
+
     /**
      * @var int the default fetch mode for this command.
      *
-     * @see http://www.php.net/manual/en/pdostatement.setfetchmode.php
+     * {@see http://www.php.net/manual/en/pdostatement.setfetchmode.php}
      */
-    public $fetchMode = \PDO::FETCH_ASSOC;
+    public int $fetchMode = PDO::FETCH_ASSOC;
+
     /**
      * @var array the parameters (name => value) that are bound to the current PDO statement.
-     *            This property is maintained by methods such as [[bindValue()]]. It is mainly provided for logging purpose
-     *            and is used to generate [[rawSql]]. Do not modify it directly.
+     *
+     * This property is maintained by methods such as {@see bindValue()}. It is mainly provided for logging purpose and
+     * is used to generate {@see rawSql}. Do not modify it directly.
      */
-    public $params = [];
+    public array $params = [];
+
     /**
      * @var int the default number of seconds that query results can remain valid in cache.
-     *          Use 0 to indicate that the cached data will never expire. And use a negative number to indicate
-     *          query cache should not be used.
+     * Use 0 to indicate that the cached data will never expire. And use a negative number to indicate query cache
+     * should not be used.
      *
      * @see cache()
      */
-    public $queryCacheDuration;
+    public ?int $queryCacheDuration = null;
+
     /**
-     * @var \Yiisoft\Cache\Dependencies\Dependency the dependency to be associated with the cached query result for this command
+     * @var Dependency the dependency to be associated with the cached query result
+     * for this command
      *
      * @see cache()
      */
-    public $queryCacheDependency;
+    public ?Dependency $queryCacheDependency = null;
 
     /**
      * @var array pending parameters to be bound to the current PDO statement.
      */
-    private $_pendingParams = [];
+    private array $pendingParams = [];
+
     /**
-     * @var string the SQL statement that this command represents
+     * @var string|null ?string the SQL statement that this command represents
      */
-    private $_sql;
+    private ?string $sql = null;
```

Comment:
For now I prefer to use null, since it tells me that the variable has no state, while if I initialize it with an empty string it will be more difficult to control, then it is preferable to leave it without typing and validate the type in the functions.